### DISCUSSION TITLE
Whether it's possible to initiate a frontend request directly within an extension

### DISCUSSION THREAD

**Discussion by alanhe421** [coder/code-server#6576] [General]
I'm curious whether it's possible to initiate a frontend request directly within an extension. I don't want to execute it in the code server, which is Node, but rather have the frontend initiate the request directly. 
From what I know so far, I can create a webview and then execute it, but this webview would certainly be visible. I don't want to add such a page; I just want to directly initiate a request. For example, if I register a command, once the user executes the command, it would then initiate such a web request.

<img width="1057" alt="image" src="https://github.com/coder/code-server/assets/9245110/b93f3f96-b8ab-48af-a8fc-e0e1dd9bc29a">

**Comment by code-asher** [comment]
There is no way to do this (that I know of), but you could patch code-server to make it possible (see your [other discussion](https://github.com/coder/code-server/discussions/6575#discussioncomment-7834443) for details).
